runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java

```diff
@@ -469,4 +488,64 @@ public void onNpcDespawned(NpcDespawned npcDespawned)
 		NPC npc = npcDespawned.getNpc();
 		npcs.remove(npc);
 	}
+
+	@Subscribe
+	public void onScriptPostFired(ScriptPostFired scriptPostFired)
+	{
+		if (scriptPostFired.getScriptId() == ScriptID.AGI_LAPS_CHECK)
+		{
+			Widget agiLapsHeader = client.getWidget(WidgetInfo.AGI_LAPS_HEADER);
+			if (agiLapsHeader != null)
+			{
+				if (agiLapsHeader.getText().equals("Agility Course Lap Counts"))
+				{
+					Widget agiCourseLapCount = client.getWidget(WidgetInfo.AGI_LAPS_COUNT);
+					String[] lapsList;
+					try
+					{
+						lapsList = agiCourseLapCount.getText().split("<br>");
+					}
+					catch (NullPointerException nullPointerException)
+					{
+						return;
+					}
```

Comment:
Why is this a try-catch? What of this code is expected to NPE and why?

Reply:
The IDE stated the getText() method may throw a NPE so I wrapped it just incase, but removed it in the updated version

Reply:
This would be because the widget can be null. You should check that the result of `client.getWidget(WidgetInfo.GENERIC_SCROLL_TEXT) != null`, and in an if statement rather than catching an NPE.